src/Models/AZ3166Device.ts

```diff
@@ -150,8 +157,82 @@ export class AZ3166Device extends ArduinoDeviceBase {
     return true;
   }
 
-  async configDeviceSettings(): Promise<boolean> {
-    const configSelectionItems = this.getConfigDeviceSettingsMainOptions();
+  async configDeviceSettings(): Promise<void> {
+    // Select device settings type
+    const deviceSettingType = await this.selectDeviceSettingType();
+
+    let credentials = '';
+    let deviceSettingsType = '';  // For log info
+    switch (deviceSettingType) {
+      case ConfigDeviceOptions.CRC:
+        await this.generateCrc(this.channel);
+        return;
+      case ConfigDeviceOptions.ConnectionString:
+        // Get device connection string
+        credentials = await this.getDeviceConnectionString();
+
+        deviceSettingsType = 'device connection string';

```

Comment:
>deviceSettingsType = 'device connection string'; [](start = 8, length = 48)

Please use the enum string you've already defined for it.

Reply:
Fixed.